include/envoy/stats/stats.h

```diff
@@ -23,7 +23,7 @@ class Metric {
   /**
    * Returns the full name of the Metric.
    */
-  virtual const std::string& name() const PURE;
+  virtual const std::string name() const PURE;
```

Comment:
do you need to make this change in this PR? I understand why you'd need to make it in the SymbolTable PR.

Ditto for the other changes where you were returning a reference and now you are returning a copy.

Reply:
Because `name_` in the RawStatData is a `char[]`, I think it might make more sense to make this return a `string_view` everywhere.

Reply:
Right, makes sense. In the context of this PR, string_view is the right option. You'll have to return string-by-value if/when we move to SymbolTable. 

The advantage of your approach here (returning string-by-value) is that it will work for SymbolTable. Switching callers to take string_view  is optimal in this PR, but would actually would be the *wrong* direction to go if we are heading to SymbolTable. I'm hopeful the compiler will tell us if we write:
```
  absl::string_view_name = stat->name();
```

if name() returns a string-by-value. If not, in the SymbolTable PR, you can just rename the method (even temporarily) to force you to examine all methods via compiler errors.

So I think my feeling is you should do the right thing or this PR and return a string_view.

Reply:
I want to keep this as-is, since we might have to convert this back into a string at one of the aforementioned interfaces/callsites, and since an incumbent symbol_table PR will play more cleanly with a string interface. Converting this to a string_view everywhere means the reworking of maps and structures which store strings now to store `container<string_view, _, StringViewHash_, StringViewCompare_>`, and they'll just end up churning again into `container<StatNamePtr, _, StatNamePtrHash_, StatNamePtrCompare_>`.